compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUPromoteMatmulToFitMMA.cpp

```diff
@@ -106,8 +97,48 @@ class LLVMGPUPromoteMatmulToFitMMAPass final
     });
 
     IRRewriter rewriter(ctx);
-    for (auto op : candidates) {
-      padWithZeroValue(rewriter, op, targetIterType, nofold);
+    for (linalg::LinalgOp op : candidates) {
+      SmallVector<int64_t> padMultiples(op.getNumLoops(), 1);
+      auto config = dyn_cast_or_null<IREE::GPU::LoweringConfigAttr>(
+          getLoweringConfig(op));
+      if (config) {
+        switch (targetDimensions) {
+        case LLVMGPUMatmulPadOption::ParallelDims:
+          padMultiples = config.getStaticTilingLevelSizes(
+              (unsigned)IREE::GPU::TilingLevel::Workgroup, op);
+          break;
+        case LLVMGPUMatmulPadOption::ReductionDims:
+          padMultiples = config.getStaticTilingLevelSizes(
+              (unsigned)IREE::GPU::TilingLevel::Reduction, op);
+          break;
+        default:
+          llvm_unreachable("Unexpected target dimensions");
+          break;
```

Comment:
nit: we are avoiding llvm_unreachable in this case because it could potentially lead to UB. See https://github.com/iree-org/iree/pull/8521 for more details

```suggestion
          assert(false && "Unexpected target dimensions");
          break;
```

Reply:
Oh wow, thank you for pointing this out!

Reply:
At the end of that thread the conclusion is that llvm_unreachable should be fine to use after all? https://github.com/iree-org/iree/pull/8521#issuecomment-1565084742

Reply:
interesting. I'm going to just use assert(false) for consistency since there was another use of assert(false) before.